### PR TITLE
place hq export in export module

### PR DESCRIPTION
**Pull request by edgardoh**
This moves the "do high quality resampling during export" setting from config option to the export module.

Addresses #2269

### CODE REVIEW

**Comment by TurboGit** [comment]
The plugins/lighttable/export/high_quality_processing setting is also referenced in darkroom.c and lua/format.c. So we cannot remove it without fixing those other places.

**Comment by edgardoh** [comment]
Yes, but is only read, not set there. And the rest of the settings already
are in the export module.

**Comment by TurboGit** [comment]
Ok. But since the preference is removed we also need to remove those read, no?

**Comment by edgardoh** [comment]
What this PR does is to change the place where this setting is modified by
the user, nothing else. The preference is still stored in the same place
and read from the same place, but now instead of a check box in config we
have a combo box in the export module. All other export settings are there,
so this seems reasonable to me.

If there's any change in any functionality that means that I have missed
something and I'll fix it.

Full disclosure, this setting is now affected by the export presets, that
can be considered as a change in functionality, but I think is a good
thing, and consistent with all other settings.

File: data/darktableconfig.xml.in [hunk 1]
```diff
@@ -1230,13 +1230,6 @@
     <shortdescription>do high quality processing for slideshow</shortdescription>
     <longdescription>same option as for export, but applies to slideshow.</longdescription>
   </dtconfig>
-  <dtconfig prefs="core" section="quality">
```
Comment (TurboGit) [inline data/darktableconfig.xml.in]: This should be <dtconfig> and we need to keep this entry. This will be an hidden pref.

**Comment by TurboGit** [comment]
I see, but then we should not remove the pref, we just need to make it hidden.

**Comment by edgardoh** [comment]
I have restored the config code, but I don't know how to hide it.

**Comment by TurboGit** [comment]
To hide it just remove the prefs and section attribute of the opening \<dtconfig\> node.

**Comment by edgardoh** [comment]
Done.

**Comment by TurboGit** [comment]
I'm still not sure about the pref used in for example the slideshow.

We indeed read it to either export in full-res or not. But now this settings depends on the last change the export module. This is not very intuitive and I don't think documenting:

- to have full-res export in slideshow first go to export module and change the corresponding setting otherwise the last value changed here or from a preset will be used.

So, what about:

- keeping the pref as it was before and used it for the slideshow
- use the new setting in the export module to control export only and do not update the pref

?

**Comment by edgardoh** [comment]
The slide show is not using this variable, is using
"plugins/slideshow/high_quality". No idea where this is set, but it has
nothing to do with this change.

**Comment by junkyardsparkle** [comment]
I've had some thoughts before about making this setting more flexible, possibly by being conditional on the degree of downscaling during export. There are good reasons (mostly related to noise reduction) for working on full size if the final export is only being scaled by a small factor, For larger scale factors, it becomes less important (as noise reduction itself does), and applying the final sharpening to the already scaled image makes more sense.

Making the setting part of export presets already helps, since presets also usually set output dimensions. Still, there are cases where a "smart" calculation of actual pixel scaling, taking cropping into account, would be better. A heavily cropped image being exported at "half size" dimensions may not actually be scaled very much, for instance. Anyway, I'm just presenting the idea. :)

**Comment by TurboGit** [comment]
@edgardoh : I see I was confused :) sorry, I'll test that now.

**Comment by TurboGit** [comment]
All good indeed. Thanks.

**Pull request merged**